src/components/interactive-builder/modals/question/question-form/rendering-types/inputs/workspace-launcher/workspace-launcher.component.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TextInput } from '@carbon/react';
+import { useFormField } from '../../../../form-field-context';
+
+const WorkspaceLauncher: React.FC = () => {
+  const { t } = useTranslation();
+  const { formField, setFormField } = useFormField();
+
+  const { buttonLabel, workspaceName } = useMemo(
+    () => ({
+      buttonLabel: formField.questionOptions?.buttonLabel ?? '',
+      workspaceName: formField.questionOptions?.workspaceName ?? '',
+    }),
+    [formField.questionOptions?.buttonLabel, formField.questionOptions?.workspaceName],
+  );
+
+  const handleButtonLabelChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const updatedQuestion = {
+        ...formField,
+        questionOptions: { ...formField.questionOptions, buttonLabel: event.target.value },
+      };
+      setFormField(updatedQuestion);
+    },
+    [formField, setFormField],
+  );
```

Comment:
Having `formField` and `setFormField` as dependencies could cause some unexpected behavior since `setFormField` updates `formField`. See here - https://github.com/openmrs/openmrs-esm-form-builder/blob/d80c22868c17a39df4838563947eeed81c9f9ffb/src/components/interactive-builder/modals/question/question-form/question-types/inputs/obs/obs-type-question.component.tsx#L46 on how to call `setFormField` in a way that you don't need the `formField` object